clients/client-ec2/src/EC2Client.ts

```diff
@@ -3588,15 +3594,26 @@ export class EC2Client extends __Client<
    */
   readonly config: EC2ClientResolvedConfig;
 
+  private getDefaultHttpAuthSchemeParametersProvider() {
+    return defaultEC2HttpAuthSchemeParametersProvider;
+  }
+
+  private getIdentityProviderConfigProvider() {
+    return async (config: EC2ClientResolvedConfig) =>
+      new DefaultIdentityProviderConfig({
+        "aws.auth#sigv4": config.credentials,
+      });
+  }
```

Comment:
private methods should be at the bottom of a class

Reply:
Got it. I think this will require intercepting a different code section the smithy-typescript.

Currently it intercepts `ClientPropertiesCodeSection`, but I'm thinking for the private methods, it should intercept `ClientBodyExtraCodeSection`.

Reply:
See: https://github.com/smithy-lang/smithy-typescript/pull/1140